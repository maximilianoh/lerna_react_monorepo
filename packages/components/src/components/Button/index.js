import React from 'react';
import PropTypes from 'prop-types';

function Button({ children, id, onClick, type }) {
  return (
    <button
      id={id}
      onClick={onClick}
      type={type}
    >
      {children}
    </button>
  );
}

Button.defaultProps = {
  id: null,
  onClick: null,
  type: 'button',
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.oneOf(['submit', 'button', 'reset']),
};

export default Button;