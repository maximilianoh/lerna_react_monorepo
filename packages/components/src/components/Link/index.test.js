import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';

test('Button primary enabled', () => {
  const component = renderer.create(
    <Button
      type="button"
    >
      soy un boton
    </Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
