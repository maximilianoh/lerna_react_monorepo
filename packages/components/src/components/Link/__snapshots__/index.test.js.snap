// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button primary enabled 1`] = `
<a
  id={null}
  onClick={null}
>
  <div
    id="address_label"
  >
    Enter your address
  </div>
  <input
    aria-labeledby="address_label"
  />
  soy un boton
</a>
`;
